### PR TITLE
Add jsx-wrap-multilines

### LINKED ISSUE #5: New Rule: jsx-wrap-multilines
Wrap JSX tags in parentheses when they span more than one line.

### PR DESCRIPTION
**Pull request by enoshixi**
Fixes #5

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/tslint-react](https://github.com/palantir/tslint-react), @enoshixi! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: test/rules/jsx-wrap-multilines/test.tsx.lint [hunk 1]
```diff
@@ -0,0 +1,42 @@
+test/rules/jsx-wrap-multilines/test.tsx.lint: Failed!
+Expected (from .lint file)
+Actual (from TSLint)
```
Comment (adidahiya) [inline test/rules/jsx-wrap-multilines/test.tsx.lint]: can you please remove the above 3 lines from this file?

File: test/rules/jsx-wrap-multilines/test.tsx.lint [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+);
+
+const badSelfClosingMultiline = <div
+                                ~~~~~
+    className="my-class"
+~~~~~~~~~~~~~~~~~~~~~~~~~
+    tabIndex={-1}
+~~~~~~~~~~~~~~~~~~
+/>;
+~~   [Multiline JSX elements must be wrapped in parenthesis]
```
Comment (adidahiya) [inline test/rules/jsx-wrap-multilines/test.tsx.lint]: how about a case like the following?

```ts
const foo = (<div
    className="bar"
/>);
```

my opinionated stance on this is that a new line should be required after the opening parenthesis and before the closing one, so the above example would be invalid. you'd need to make the error message very clear though

File: src/rules/jsxWrapMulitlinesRule.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE = "Multiline JSX elements must be wrapped in parenthesis";
```
Comment (adidahiya) [inline src/rules/jsxWrapMulitlinesRule.ts]: This should say "parentheses" because they're plural.

File: src/rules/jsxWrapMulitlinesRule.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/**
```
Comment (adidahiya) [inline src/rules/jsxWrapMulitlinesRule.ts]: your file name does not match the walker name. the file should be `jsxWrapMultilineRule.ts`, which makes the rule name `jsx-wrap-multiline`.

**Comment by adidahiya** [comment]
@enoshixi looks like the build is failing on master after merging this. Can you please

1. enable CircleCI on your fork
2. investigate the build failures here? https://circleci.com/gh/palantir/tslint-react/88?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

thanks.

**Comment by enoshixi** [comment]
Will do.

**Pull request merged**